dot/telemetry/mailer_test.go

```diff
@@ -20,35 +20,42 @@ import (
 	"github.com/ChainSafe/gossamer/lib/common"
 	"github.com/ChainSafe/gossamer/lib/genesis"
 	"github.com/gorilla/websocket"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-var upgrader = websocket.Upgrader{}
-var resultCh chan []byte
+func newTestMailer(t *testing.T, handler http.HandlerFunc) (mailer *Mailer) {
+	t.Helper()
 
-func TestMain(m *testing.M) {
-	// start server to listen for websocket connections
-	upgrader.CheckOrigin = func(r *http.Request) bool { return true }
-	http.HandleFunc("/", listen)
-	go http.ListenAndServe("127.0.0.1:8001", nil)
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", handler)
 
-	time.Sleep(time.Millisecond)
-	// instantiate telemetry to connect to websocket (test) server
-	var testEndpoints []*genesis.TelemetryEndpoint
+	srv := httptest.NewServer(mux)
+	t.Cleanup(func() {
+		srv.Close()
+	})
```

Comment:
nit do you need this? Maybe `httptest` handles it for you? :thinking: I have no idea, just being curious here

Reply:
Yes we need, `httptest` does not closes after test ends (https://pkg.go.dev/net/http/httptest#example-Server)